make-release.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+ENAME=$0
+
+usage()
+{
+cat << EOF
+Usage: $ENAME OPTION
+  -x Clean git repository before preparing release
+EOF
+}
+
+while getopts "x" opt; do
+	echo foo $opt
+	case "$opt" in
+	x )
+		git clean -ffdx
+		;;
+	? )
+		usage
+		exit 1
+		;;
+	esac
+done
+
+logfile=release.$(date +%y-%m-%d-%T).log
+echo "Logging to ${logfile}"
+
+# TODO: Handle tagging?
+
+if [ ! -e ./configure ]; then
+	echo "Configure missing. Did you bootstrap?"
+fi
+
+config_opts=""
+if ./configure --help | grep -- --enable-maintainer-mode > /dev/null; then
+	config_opts+=" --enable-maintainer-mode";
+fi
+
+echo "Running configure with options '${config_opts}'"
+if ! ./configure $config_opts >> ${logfile} 2>&1; then
+	echo "Configuration failed. Aborting"
+	exit 1
+fi
+
+if [ ! -e Makefile ]; then
+	echo "Makefile missing. Aborting"
+	exit 1
+fi
+
+# TODO - fixme
+# This repo has a weird dependency between the test
+# and src directories to modify auxv.o
+make >> ${logfile} 2>&1
+
+
+echo "Running make distcheck"
+make distcheck >> ${logfile} 2>&1
```

Comment:
>Point to notice, configure is only run twice (first with --help to check for the maintainer mode option), second to perform the actual configuration.

Oops.  My bad.

> The "make" overcomes a dependency problem before running "make distcheck". Talking with Gabriel yesterday, there is a troublesome rule which sneaks into the src/.deps folder and modifies an object file and builds a secondary shared library.

Good point.  We can live with the duplicated make + TODO, but could you elaborate the TODO, please?